components/camel-swagger-java/src/main/java/org/apache/camel/swagger/RestSwaggerSupport.java

```diff
@@ -345,4 +353,36 @@ private static void setupCorsHeaders(RestApiResponseAdapter response, Map<String
         response.setHeader("Access-Control-Max-Age", maxAge);
     }
 
+    private void setupXForwardedHeaders(Swagger swagger, Map<String, Object> headers) {
+
+        String host = (String) headers.get(HEADER_HOST);
+        if(host != null && host.length() > 0) {
+            swagger.setHost(host);
+        }
+
+        String forwardedPrefix = (String) headers.get(HEADER_X_FORWARDED_PREFIX);
+        if (forwardedPrefix != null && forwardedPrefix.length() > 0) {
+            swagger.setBasePath("/" + forwardedPrefix.replace("/", "") + swagger.getBasePath());
```

Comment:
I think this might lead to issues if the `forwardedPrefix` is `"a/b"` then `basePath` would be set to `"ab"` which is probably not what we want.

Perhaps I'm not following the logic here is the idea to have the resulting `basePath` equal `X-Forwarded-Prefix` + `basePath`?

Perhaps we could use something like:

```java
String prefixedBasePath = "/" + URISupport.stripPrefix(forwardedPrefix, "/") + "/" URISupport.stripPrefix(swagger.getBasePath(), "/");
```